internal/executor/taskfilter/taskfilter.go

```diff
@@ -18,14 +18,64 @@ func MatchExactTask(desiredTaskName string) TaskFilter {
 		var filteredTasks []*api.Task
 
 		for _, task := range tasks {
-			if strings.EqualFold(desiredTaskName, task.Name) {
-				// Clear task's dependencies
-				task.RequiredGroups = task.RequiredGroups[:0]
+			// Ensure that this task's name matches with the name we're looking for
+			desiredTaskNameLower := strings.ToLower(desiredTaskName)
+			taskNameLower := strings.ToLower(task.Name)
 
-				filteredTasks = append(filteredTasks, task)
+			if !strings.HasPrefix(desiredTaskNameLower, taskNameLower) {
+				continue
 			}
+
+			// In case we're looking for a task with specific labels — extract them and ensure they all match
+			desiredLabels := extractLabels(strings.TrimPrefix(desiredTaskNameLower, taskNameLower))
+
+			var actualLabels []string
+			if task.Metadata != nil {
+				actualLabels = task.Metadata.UniqueLabels
+			}
+
+			if !containsAll(desiredLabels, actualLabels) {
```

Comment:
This should be also fixed with the below suggestion

```suggestion
			if !containsAll(actualLabels, desiredLabels) {
```